client/app/states/login/login.state.js

```diff
@@ -101,10 +132,26 @@ function StateController ($window, $state, Text, RBAC, API_LOGIN, API_PASSWORD,
     })
   }
 
-  function getBrand () {
+  function getProductInfo () {
     CollectionsApi.query('product_info').then((response) => {
       vm.brandInfo = response.branding_info
       $rootScope.favicon = vm.brandInfo.favicon
+      vm.authenticationInfo = response.authentication
+      vm.extAuthMode        = getExtAuthMode()
+      if (vm.extAuthMode == 'oidc') {
+        if (vm.oidcInitiatedLogin) {
+          vm.oidcInitiatedLogin = false
+          let idp_access_token = $cookies.get(oidc_access_token.name)
+          if (idp_access_token && idp_access_token != '') {
+            vm.access_token = idp_access_token
+            return AuthenticateUser()
+          }
+        } else {
+          if (vm.authenticationInfo.sso_enabled) {
+            return new Promise( () => { initiateOidcLogin() })
```

Comment:
possibly a bug .. this returns a promise that never resolves or rejects

is this to run `initiateOidcLogin` asynchronously later? If so, maybe `$timeout(initiateOidcLogin)` would be clearer?

Reply:
triggers the IDP login when sso is enabled, $timeout(initiateOidcLogin) worked great too 👍